src/Public/OpenApi.ps1

```diff
@@ -146,32 +302,49 @@ function Get-PodeOpenApiDefinition {
         [string]
         $RouteFilter = '/*',
 
+        [Parameter()]
         [switch]
         $RestrictRoutes
     )
 
-    $Title = Protect-PodeValue -Value $Title -Default $PodeContext.Server.OpenAPI.Title
-    if ([string]::IsNullOrWhiteSpace($Title)) {
-        throw 'No Title supplied for OpenAPI definition'
+    $meta = @{
+        RouteFilter    = $RouteFilter
+        RestrictRoutes = $RestrictRoutes
     }
-
-    $Version = Protect-PodeValue -Value $Version -Default $PodeContext.Server.OpenAPI.Version
-    if ([string]::IsNullOrWhiteSpace($Version)) {
-        throw 'No Version supplied for OpenAPI definition'
+    if ($RestrictRoutes) {
+        $meta = @{
+            RouteFilter    = $RouteFilter
+            RestrictRoutes = $RestrictRoutes
+        }
+    } else {
+        $meta = @{}
+    }
+    if ($Title) {
+        $meta.Title = $Title
+    }
+    if ($Version) {
+        $meta.Version = $Version
+    }
+    if ($Description) {
+        $meta.Description = $Description
     }
 
-    $Description = Protect-PodeValue -Value $Description -Default $PodeContext.Server.OpenAPI.Description
+    $oApi = Get-PodeOpenApiDefinitionInternal  -MetaInfo $meta
```

Comment:
Missing the Protocol/Address/EndpointName parameters from the original
```powershell
-Protocol $WebEvent.Endpoint.Protocol `
-Address $WebEvent.Endpoint.Address `
-EndpointName $WebEvent.Endpoint.Name `
```

Reply:
it's not needed.
 Get-PodeOpenApiDefinitionInternal use -MetaInfo for that

Reply:
I don't think we have to keep the back compatibility for the internal function

Reply:
Funnily enough, looking at the origina internal function, I don't think Protocol/Address were being used! However, even in the new internal function the `$EndpointName` parameter is still being used, and isn't being sourced from `$MetaInfo`. The routes created further up to get the OpenAPI definition do also pass these parameters to te internal function as well.